### PR TITLE
Feat: switch to fastapi

### PR DESCRIPTION
**Pull request by julianmarmier**
- Fully switched to FastAPI and Celery as the backend
- Restructured the project to try to better separate the server part from the ML part (let me know/change if you have better ideas)
- Added docker support across the stack (message broker [Redis], task queue [Celery beat + worker], ASGI server [FastAPI], frontend [React/Yarn]

@Yingrjimsch @ChungYi347 @chenzhutian let me know what you think of the revised file structure and if you have any comments you would like to add!

### CODE REVIEW

File: server/celery/celery.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        )
+    )
+
+    return celery_app
+
+
+os.environ["OPENAI_API_KEY"] = apikey
+os.environ["OPENAI_ORGANIZATION"] = organization
+
+os.makedirs("files", exist_ok=True)
```
Comment (Yingrjimsch) [inline server/celery/celery.py:32]: This is done here and in backend can we init it once?

File: server/routes.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+async def default_page():
+    return RedirectResponse("/docs")
+
+
+@router.get("/transcript")
+async def run_transcription():
+    return FileResponse("/files/transcript.vtt")
+
+
+@router.get("/sub")
```
Comment (Yingrjimsch) [inline server/routes.py:24]: Can we rename that to subtitle and make /infer a little clearer?

**Review by Yingrjimsch — COMMENTED** [review]
Looks good to me for now. I'd be happy if we could rename the docker-compose backend from web to backend and frontend from frontend to webapp so it's the same as the repo.

Another thing is that we have to change the filestore a little bit. Every uploaded file should get it's own unique folder in files (with timestamp or UUID). Inside this folder the structure should look always the same so at the end it looks something like
- files
  - UUID
   - documents
     - transcript.vtt
     - playbyplay.txt
     - transcript.txt
   - video.mp4
   - video_transcript.mp4

File: server/celery/celery.py (hunk 1 above)
Comment (julianmarmier) [reply server/celery/celery.py:32]: I'll try initializing it only in backend

File: server/routes.py (hunk 2 above)
Comment (julianmarmier) [reply server/routes.py:24]: Yep! Maybe rename `/infer` to `/ask` and add it as a Celery task?

**Comment by julianmarmier** [comment]
Ok, so I'm going to update the API a bit to add a video UUID but in a separate branch

**Comment by Yingrjimsch** [comment]
> Ok, so I'm going to update the API a bit to add a video UUID but in a separate branch

In main there is a files/12345 as example

**Pull request merged**